app/agents_v2/primary_agent/agent.py

```
@@ -176,434 +81,61 @@ async def run_primary_agent(state: PrimaryAgentState) -> AsyncIterator[AIMessage
+                await anyio.sleep(0.005)  # Reduced delay for smoother streaming
+
+            parent_span.set_status(Status(StatusCode.OK))
+
+        except Exception as e:
+            logger.exception("Error in run_primary_agent: %s", e)
+            if 'parent_span' in locals() and parent_span.is_recording():
                 parent_span.record_exception(e)
                 parent_span.set_status(Status(StatusCode.ERROR, str(e)))
-            yield AIMessageChunk(content=f"An internal error occurred: {str(e)}", role="assistant") 
\ No newline at end of file
+            yield AIMessageChunk(content=f"I'm sorry, an unexpected error occurred. Please try again later.", role="assistant") 
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)